### PR TITLE
Add explicit Access ODBC decimal type mapping

### LINKED ISSUE #3709: [Scaffold] Investigate why `exclude-schemas` option breaks Access scaffold
Hi, I am trying to generate a scaffold from a older MS Access database. For this, I've created the following JSON configuration file:

```{
  "general": {
    "provider": "Access",
    "connection": "Driver={Microsoft Access Driver (*.mdb, *.accdb)};Dbq=C:\\Users\\kai\\Documents\\T2000_b\\IKS-DB.mdb;Persist Security Info=False;",
    "additional-connection": "Provider=Microsoft.ACE.OLEDB.12.0;Data Source=C:\\Users\\kai\\Documents\\T2000_b\\IKS-DB.mdb;Persist Security Info=False",
    "output": "C:\\Users\\kai\\RiderProjects\\DBAccess\\DBAccess\\",
    "overwrite": true
  },
    "schema" : {
      "include-tables": [{"name":  "HOSTS"},{"name": "FNT-All"}],
      "exclude-schemas": ["Check","Server","installiert","nicht abgebaut","nicht in FNT"]
    },
  "code": {
    "single-file": true
  }
}
```
and I'm invoking it with
```
C:\Users\kai> dotnet-linq2db.exe scaffold -p Access -c "Driver={Microsoft Access Driver (*.mdb, *.accdb)};Dbq=C:\Users\kai\Documents\T2000_b\IKS-DB.mdb;Persist Security Info=False;" --additional-connection "Provider=Microsoft.ACE.OLEDB.12.0;Data Source=C:\Users\kai\Documents\T2000_b\IKS-DB.mdb;Persist Security Info=False" -i C:\Users\kai\RiderProjects\DBAccess\DBAccess\table.json
```

It fails with the following error message:
```
Unhandled exception: ColumnType not provided by schema for table Check, Server, installiert, nicht abgebaut, nicht in FNT
   at LinqToDB.Schema.LegacySchemaProvider.ParseColumn(SqlObjectName tableName, ColumnSchema column)
   at LinqToDB.Schema.LegacySchemaProvider.ParseTable(TableSchema table, SchemaObjects loadedObjects)
   at LinqToDB.Schema.LegacySchemaProvider.ParseSchema(DatabaseSchema schema, SchemaObjects loadedObjects)
   at LinqToDB.Schema.LegacySchemaProvider..ctor(DataConnection connection, SchemaOptions options, ILanguageProvider languageProvider)
   at LinqToDB.CommandLine.ScaffoldCommand.Scaffold(ScaffoldOptions settings, ScaffoldInterceptors interceptors, String provider, String providerLocation, String connectionString, String additionalConnectionString, String output, Boolean overwrite)
   at LinqToDB.CommandLine.ScaffoldCommand.Execute(CliController controller, String[] rawArgs, Dictionary`2 options, IReadOnlyCollection`1 unknownArgs)
   at LinqToDB.CommandLine.CliController.Execute(String[] args)
   at LinqToDB.Tools.Program.Main(String[] args)
```
which is interesting, because I've checked the schemas are not used in the tables. Furthermore I'm using the Linq2DB for LINQPad 7 (v.4.1.1.0) and created a connection to the database. Here it works without a problem. I hope you can help me.

Thanks in advance,

Kai

_Originally posted by @Delta-VII in https://github.com/linq2db/linq2db/discussions/3706_

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #3709

Add explicit DECIMAL type mapping, ensure we will not have such faults in future

### CODE REVIEW

File: Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.cs [hunk 1]
```diff
@@ -201,6 +201,7 @@ private TypeMapping MapType(DatabaseType databaseType)
 			mapping = _interceptors.GetTypeMapping(databaseType, _languageProvider.TypeParser, mapping);
 			if (mapping == null)
 			{
+				// TODO: use logger
```
Comment (MaceWindu) [inline Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.cs:204]: this is a backlog task to add logger interface to scaffold

File: Source/LinqToDB.Tools/Schema/LegacySchemaProvider.cs [hunk 2]
```diff
@@ -470,10 +470,11 @@ private Column ParseColumn(SqlObjectName tableName, ColumnSchema column)
 					column.ColumnType = "NUMERIC";
 				}
 				else
-					throw new InvalidOperationException($"ColumnType not provided by schema for table {tableName}");
+					// TODO: use logger
+					Console.Error.WriteLine($"ColumnType not provided by schema for column {tableName}.{column.ColumnName}");
```
Comment (MaceWindu) [inline Source/LinqToDB.Tools/Schema/LegacySchemaProvider.cs:474]: convert to non-fatal error

File: Source/LinqToDB/DataProvider/Access/AccessODBCSchemaProvider.cs [hunk 3]
```diff
@@ -87,7 +87,7 @@ protected override List<ColumnInfo> GetColumns(DataConnection dataConnection, Ge
 					Name        = c.Field<string>("COLUMN_NAME")!,
 					IsNullable  = c.Field<short> ("NULLABLE") == 1,
 					Ordinal     = Converter.ChangeTypeTo<int>(c["ORDINAL_POSITION"]),
-					DataType    = dt?.TypeName,
+					DataType    = dt?.TypeName ?? typeName,
```
Comment (MaceWindu) [inline Source/LinqToDB/DataProvider/Access/AccessODBCSchemaProvider.cs:90]: fallback to raw type name if type mapping missing

File: Source/LinqToDB/DataProvider/Access/AccessODBCSchemaProvider.cs [hunk 4]
```diff
@@ -201,6 +201,17 @@ protected override List<DataTypeInfo> GetDataTypes(DataConnection dataConnection
 				});
 			}
 
+			if (dts.All(dt => dt.TypeName != "DECIMAL"))
```
Comment (MaceWindu) [inline Source/LinqToDB/DataProvider/Access/AccessODBCSchemaProvider.cs:204]: decimal type mapping disappeared from provider for unknown reason...

**Review by sdanyliv — APPROVED** [review]

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**